### PR TITLE
[MRG] Fix mdd loss

### PR DESCRIPTION
**Pull request by antoinecollas**
Use equations (26)-(29) from https://arxiv.org/pdf/1904.05801

### CODE REVIEW

**Comment by antoinedemathelin** [comment]
It seems nice, but why adding the new arguments domain_criterion_t and domain_criterion_s? I feel it adds too much complexity. Maybe we can say that the original MDD paper only focus on multi classification, and we do the same here. We can almost remove the domain_criterion argument and hardset it to be crossentropy for source and custom crossentropy for target. 
Wdyt?

We can let to future PRs the generalization of MDD to other losses?

**Comment by antoinecollas** [comment]
I agree, I change this

File: skada/deep/_adversarial.py [hunk 1]
```diff
@@ -421,11 +429,13 @@ def forward(
         pseudo_label_s = torch.argmax(y_pred_s, axis=-1)
         pseudo_label_t = torch.argmax(y_pred_t, axis=-1)
 
-        disc_loss_src = self.disc_criterion_(disc_pred_s, pseudo_label_s)
-        disc_loss_tgt = self.disc_criterion_(disc_pred_t, pseudo_label_t)
+        disc_criterion_s_ = torch.nn.CrossEntropyLoss()
+        disc_loss_s = disc_criterion_s_(disc_pred_s, pseudo_label_s)
+        disc_criterion_t_ = ModifiedCrossEntropyLoss()
+        disc_loss_t = disc_criterion_t_(disc_pred_t, pseudo_label_t)
```
Comment (antoinedemathelin) [inline skada/deep/_adversarial.py:435]: Is it ok to instantiate functions in the forward pass (I mean from the computational efficiency point of view)?
Maybe it is better to define the criterions in the init as attributes and then use the same in forward?

**Comment by antoinedemathelin** [comment]
Ok, I just have one comment, but the rest is ok for me.

File: skada/deep/_adversarial.py [hunk 2]
```diff
@@ -370,6 +371,22 @@ def CDAN(
...
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, input, target):
+        """Compute the modified CrossEntropyLoss"""
+        prob = F.softmax(input, dim=-1)
+        prob = prob[..., target]
+        log_one_minus_prob = torch.log(1 - prob)
```
Comment (tomMoral) [inline skada/deep/_adversarial.py:386]: Maybe use this one: https://pytorch.org/docs/stable/generated/torch.log1p.html
